helper_windows_test.go

```diff
@@ -49,18 +49,28 @@ func incrementCounterInCache() []string {
 	return []string{command}
 }
 
-func sleep(seconds uint) []string {
+func goEnv() []string {
 	return []string{
-		"ping 127.0.0.1 -n " + strconv.Itoa(int(seconds+1)) + " > nul",
+		"go env",
+		"set",
+		"where go",
+		"go version",
 	}
 }
 
-func goRun(goFile string) []string {
+func sleep(seconds uint) []string {
 	return []string{
-		"go run \"" + goFile + "\"",
+		"ping 127.0.0.1 -n " + strconv.Itoa(int(seconds+1)) + " > nul",
```

Comment:
heh, interesting.  I'm not too sure where this is used, but is there any reason not to use `time.Sleep`?  I'm guessing this is a helper for doing command running.

Reply:
Yes, the `sleep` function is something for generating a task command to sleep - so something that executes in the context of the task. Some of the tests use this, for example to generate a task the sleeps for 5 seconds, but has a `maxRunTime` of 3 seconds, to check that the maxRunTime property is respected and that a task that runs too long resolves correctly. The implementation for sleeping is different on windows and macOS and linux, and therefore they implement this function differently. The most compatible way to sleep from a batch script on windows is this way - there are simpler ways (like a sleep command) on more recent versions of Windows, but the ping method works across all Windows versions that we care about.

Reply:
i had a feeling that might be it.  Gotta love windows for giving you opportunities for creative problem solving!